libs/payments/ui/src/lib/client/components/FluentLocalizationProvider.tsx

```diff
@@ -4,36 +4,50 @@
 
 'use client';
 
-import React, { useState, useEffect } from 'react';
-import { ReactLocalization, LocalizationProvider } from '@fluent/react';
+import { FluentBundle, FluentResource } from '@fluent/bundle';
 import {
-  LocalizerClient,
-  LocalizerBindingsClient,
-} from '@fxa/shared/l10n/client';
+  LocalizationProvider,
+  MarkupParser,
+  ReactLocalization,
+} from '@fluent/react';
 
 export function FluentLocalizationProvider({
+  fetchedMessages,
   children,
 }: {
+  fetchedMessages: Record<string, string>;
   children: React.ReactNode;
 }) {
-  const [l10n, setL10n] = useState<ReactLocalization>();
-
-  useEffect(() => {
-    const setLocalization = async () => {
-      const locale = navigator.language;
-      const bindings = new LocalizerBindingsClient();
-      const localizerClient = new LocalizerClient(bindings);
-      const { l10n } = await localizerClient.setupReactLocalization(locale);
-
-      setL10n(l10n);
-    };
-
-    setLocalization();
-  }, []);
-
-  if (!l10n) {
-    return;
-  }
+  const bundles: FluentBundle[] = [];
+
+  Object.keys(fetchedMessages).forEach((locale) => {
+    const source = fetchedMessages[locale];
+    if (source) {
+      const bundle = new FluentBundle(locale, {
+        useIsolating: false,
+      });
+      const resource = new FluentResource(source);
+      bundle.addResource(resource);
+      bundles.push(bundle);
+    }
+  });
+
+  // To enable server-side rendering, all tags are converted to plain text nodes.
+  // They will be upgraded to regular HTML elements in the browser:
+  const parseMarkup: MarkupParser | undefined =
+    // Ignored for test coverage, since `document` is mocked (and thus defined)
+    // in unit tests:
+    /* c8 ignore next 7 */
```

Comment:
Interesting, should this be `7` or `8`?
I think +7 lines is up until (and including) the `]`, but not `: undefined;` (if I'm maths-ing correctly). Or switch to [`/* c8 ignore start */ ... /* c8 ignore stop */`](https://github.com/bcoe/c8/blob/main/README.md#ignoring-all-lines-until-told) if we want to be more explicit.